ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_layers.py

```diff
@@ -1146,3 +1146,53 @@ def test_rfftn(
         axes=axes,
         norm=norm,
     )
+
+
+@st.composite
+def valid_overlad_and_add_function(draw):
+    dtype = draw(helpers.get_dtypes(kind="float", full=False, key="dtype"))
+    x_dim = draw(
+        helpers.get_shape(
+            min_dim_size=2, max_dim_size=100, min_num_dims=2, max_num_dims=4
+        )
+    )
+    x = draw(
+        helpers.array_values(
+            dtype=dtype[0],
+            shape=tuple(x_dim),
+            min_value=-1e-10,
+            max_value=1e10,
+        )
+    )
+    frame_step = draw(helpers.ints(min_value=1, max_value=1e6))
+
+    return dtype, x, frame_step
+
+
+@handle_test(
+    fn_tree="functional.ivy.experimental.overlap_and_add",
+    dtype_x_and_args=valid_overlad_and_add_function(),
+    ground_truth_backend="tensorflow",
+    test_with_out=st.just(False),
+)
+def test_overlap_and_add(
+    *,
+    dtype_x_and_args,
+    test_flags,
+    backend_fw,
+    fn_name,
+    on_device,
+    ground_truth_backend,
+):
+    input_dtype, x, frame_step = dtype_x_and_args
+    helpers.test_function(
+        ground_truth_backend=ground_truth_backend,
+        input_dtypes=input_dtype,
```

Comment:
Seems like those are the lines causing the error

```suggestion
    on_device,
):
    input_dtype, x, frame_step = dtype_x_and_args
    helpers.test_function(
        input_dtypes=input_dtype,
```